arch/x86/core/ia32/excstub.S

```diff
@@ -121,6 +121,13 @@ SECTION_FUNC(TEXT, _exception_enter)
 	pushl	%eax             /* Save calculated ESP */
 #ifdef CONFIG_USERSPACE
 2:
+#endif
+#ifdef CONFIG_GDB_SUPPORT
```

Comment:
can we move this earlier, perhaps right next to where the co-op registers are pushed around like 104? they are only needed for debug tools as well.

Reply:
I changed, but the reason I put there was because there is some arithmetic based on $esp and as these register are conditionally pushed to stack it would require some extra `ifdef`. Solved the problem using a temporary variable